src/main/haxe/yloader/impl/js/NodeLoader.hx

```diff
@@ -0,0 +1,182 @@
+		writeData();
+		endRequest();
+	}
+
+	public function cancel():Void
+	{
+		if (clientRequest != null)
+		{
+			removeClientRequestListeners();
+			clientRequest.abort();
+		}
+		clientRequest = null;
+		response = null;
+		responseString = "";
+	}
+
+	function createClientRequest():Void
+	{
+		var url:UrlData = Url.parse(request.urlWithQuery, true);
+		if (url.protocol == "http:")
+			clientRequest = createHttpRequest(url);
+		else if (url.protocol == "https:")
+			clientRequest = createHttpsRequest(url);
+	}
+
+	function addClientRequestListeners():Void
+	{
+		clientRequest.on("error", onClientRequestError);
+	}
+
+	function removeClientRequestListeners():Void
+	{
+		clientRequest.removeAllListeners("error");
+	}
+
+	function writeData():Void
+	{
+		if (request.data == null || request.data.length < 1)
+			return;
+
+		if (clientRequest != null)
+			clientRequest.write(request.data);
+	}
+
+	function endRequest():Void
+	{
+		if (clientRequest != null)
+			clientRequest.end();
+	}
+
+	function createHttpRequest(url:UrlData):ClientRequest
+	{
+		var options:HttpRequestOptions = createOptions(url);
+		options.headers = getHeaders();
+		return Http.request(options, onRequestResponse);
+	}
+
+	function createHttpsRequest(url:UrlData):ClientRequest
+	{
+		var options:HttpsRequestOptions = createOptions(url);
+		options.headers = getHeaders();
+		return Https.request(options, onRequestResponse);
+	}
+
+	function createOptions(url:UrlData):Dynamic
+	{
+		return {
+			host: url.host,
+			port: Std.parseInt(url.port),
+			path: url.path,
+			method: Std.string(request.method)
+		};
+	}
+
+	function getHeaders():Dynamic
+	{
+		var headers:Dynamic;
+		headers = HeaderUtil.toObject(request.headers);
+		setContentLengthHeader(headers);
+		return headers;
+	}
+
+	function setContentLengthHeader(headers:Dynamic):Void
+	{
+		if (request.data == null || request.data.length < 1 || headers == null)
+			return;
+
+		if (Reflect.hasField(headers, "content-length") || Reflect.hasField(headers, "Content-Length"))
+			return;
+
+		Reflect.setField(headers, "Content-Length", Std.string(Buffer.byteLength(request.data)));
+	}
+
+	function handleResponse(responseObject:Dynamic)
+	{
+		if(onResponse == null || response == null)
+			return;
+
+		var response = getResponse(responseObject);
+		onResponse(response);
+	}
+
+	function getResponse(responseObject:Dynamic):Response
+	{
+		return new Response(isSuccess(), responseObject, response.statusCode,
+		response.statusMessage, getResponseHeaders(response));
+	}
+
+	function isSuccess():Bool
+	{
+		return response.statusCode >= 200 && response.statusCode < 400;
+	}
+
+	function getResponseHeaders(source:IncomingMessage):Array<Parameter>
+	{
+		var ret:Array<Parameter> = [];
+		for (key in source.headers.keys())
+			ret.push(new Parameter(key, source.headers.get(key)));
+
+		return ret;
+	}
+
+	function onRequestResponse(response:IncomingMessage):Void
+	{
+		this.response = response;
+		response.setEncoding("utf-8");
```

Comment:
have that string in a priavate property so it can be potentially overriden if needed